app/src/main/java/com/alexstyl/themingandroid/AppCompatThemedActivity.java

```diff
@@ -38,10 +38,10 @@ protected final void applyTheme(AppTheme theme) {
     }
 
     private void restartActivity() {
-        Intent intent = getIntent();
+        final Intent intent = getIntent();
         intent.removeCategory(Intent.CATEGORY_LAUNCHER);
-        startActivity(intent);
         finish();
+        startActivity(intent);
```

Comment:
While this might be true, I thought that someone else might encounter the same issue I did. That's why I left ordering like this based on my use case. But no prob I'll revert it back.

FWIW Using the previous ordering, Activity would just stop and never restarted again when called from within a fragment. So unless I use `recreate()` which uses no animations this was my best bet. As far as the animation goes, it makes no difference to my eyes. I Couldn't notice any emptiness or stuttering whatsoever.